src/modules/dataset/parseValueBySettingType.js

```diff
@@ -7,7 +7,7 @@ export const parseValueBySettingType = (name, value) => {
     switch (name) {
         case PERIOD_DS_DOWNLOAD:
         case PERIOD_DS_DB_TRIMMING:
-            return value ? parseInt(value) : 0
+            return value <= 0 ? 0 : parseInt(value)
```

Comment:
`value` is a string here right?  Why would it ever be `<= 0`?

Reply:
Hi @amcgee! yup `value` is a string, but when using the arrows from number input you can get a **negative integer**.  Period values should only be positive integers.

I added a console.log using `return value ? parseInt(value) : 0`  and this was the result  `{name: "periodDSDownload", value: "-1"}`

Reply:
I think this will work, oddly, because `"-1" <= 0` in weird Javascript logic.  However, I think it would be a lot clearer to do this:

```js
return Math.max(0, parseInt(value))
```

This assumes that `value` will always be a valid positive or negative number, not an arbitrary string.  If it could be an arbitrary string it would need to be:

```js
isNaN(value) ? 0 : Math.max(0, parseInt(value))
```

<img width="348" alt="Screen Shot 2020-11-02 at 4 01 39 PM" src="https://user-images.githubusercontent.com/947888/97883080-b2c1b700-1d24-11eb-85c0-5d90618edfbd.png">

Reply:
I was checking this for an arbitrary string:

![image](https://user-images.githubusercontent.com/29384664/97946429-62644c80-1d58-11eb-94ed-630a7b7a272f.png)

the thing is that when is an empty string the value is NaN, if there is no value it should be 0; so I was thinking in do it like this:

`isNaN(parseInt(value)) ? 0 : Math.max(0, parseInt(value)) `
 
![image](https://user-images.githubusercontent.com/29384664/97946279-f1249980-1d57-11eb-826b-6a06afe1ea93.png)